Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -539,41 +452,59 @@ void CGPathAddEllipseInRect(CGMutablePathRef path, const CGAffineTransform* m, C
  @Status Interoperable
 */
 void CGPathCloseSubpath(CGMutablePathRef path) {
-    if (path == NULL) {
-        return;
+    // We cannot use end figure with the built in way of closing a sub path due to
+    // an issue with reading information
+    // from a D2DGeometry. If a figure is open and any information is requested
+    // from the path, the call will fail with
+    // a bad state. Because of this, we have to ensure that any open figures are
+    // closed before continuing. The problem this
+    // causes is that the sub path is no longer being tracked via figures and we
+    // must close the sub path ourselves b
+    // drawing a line to the original point of the path.
+    if (!CGPointEqualToPoint(path->_impl.currentPoint, path->_impl.startingPoint)) {
+        CGPathAddLineToPoint(path, nullptr, path->_impl.startingPoint.x, path->_impl.startingPoint.y);
     }
-
-    _CGPathAddElement(path, kCGPathElementCloseSubpath);
+    path->endFigure(D2D1_FIGURE_END_CLOSED);
 }
 
 /**
- @Status Interoperable
+@Status Stub
+@Notes
 */
 CGRect CGPathGetBoundingBox(CGPathRef path) {
-    CGRect ret;
-
     if (path == NULL) {
-        return CGRectMake(std::numeric_limits<float>::infinity(), std::numeric_limits<float>::infinity(), 0, 0);
+        return CGRectMake(INFINITY, INFINITY, 0, 0);
     }
 
-    auto context = CGBitmapContextCreate(0, 1, 1, 1, 1, 0, 0);
-    path->_applyPath(context);
-    ret = CGContextGetPathBoundingBox(context);
-    CGContextRelease(context);
+    D2D1_RECT_F bounds;
```

Comment:
D2DRectFToCG

---
In reply to: [83932636](https://github.com/Microsoft/WinObjC/pull/1138#discussion_r83932636) [](ancestors = 83932636)